pkg/app/piped/trigger/trigger.go

```diff
@@ -219,10 +211,16 @@ func (t *Trigger) checkRepoCandidates(ctx context.Context, repoID string, cs []c
 		onOutOfSync: NewOnOutOfSyncDeterminer(t.apiClient),
 		onCommit:    NewOnCommitDeterminer(gitRepo, headCommit.Hash, t.commitStore, t.logger),
 	}
+	triggered := make(map[string]struct{})
 
 	for _, c := range cs {
 		app := c.application
 
+		// Avoid triggering multiple deployments for the same application in the same iteration.
+		if _, ok := triggered[app.Id]; ok {
+			continue
+		}
```

Comment:
Looks like the trigger which comes first has a higher priority than the one of the same application which comes after. I'm okay with that but since those candidates come from different kinds (onCommand, onCommit, onOutOfSync, onChain), maybe we need a better way to choose which one should have a higher priority base on the `c.Kind` or something.

Reply:
Yes. I understand your feeling. 👍 
Tbh, I was thinking about that. One way I was planned to use is sorting the candidate list in this function by `kind`, but after the second thought, I decided to do nothing because the passing list could be controlled by the caller. The caller can sort in the order it wants.
Maybe in the future, when we have more `kind`s we can add some kind of `priority` field to the candidate struct to decide.

Reply:
Okay, let's keep it as is and make changes to use ordering later.